File: Release/Example/Grafcet.c
#include<stdio.h>
int X0 = 1, X1 = 0, X2 = 0, X20 = 1, X21 = 0, X22 = 0, X220 = 1, X221 = 0, X222 = 0, X223 = 0, X224 = 0, X23 = 0, X230 = 1, X231 = 0, X232 = 0, X233 = 0, X234 = 0, X235 = 0, X24 = 0, X3 = 0, X4 = 0, X5 = 0, X6 = 0;
void grafcet0();
void datapath0();
void action0();
void action1();
void action2();
void grafcet2();
void datapath2();
void action3();
void action4();
void action5();
void action6();
void action20();
void action21();
void action22();
void grafcet22();
void datapath22();
void action23();
void grafcet23();
void datapath23();
void action24();
void action220();
void action221();
void action222();
void action223();
void action224();
void action230();
void action231();
void action232();
void action233();
void action234();
void action235();
void main()
{
printf("X0 = %d,X1 = %d,X2 = %d,X3 = %d,X4 = %d,X5 = %d,X6 = %d\n",X0 ,X1 ,X2 ,X3 ,X4 ,X5 ,X6 );
while(1)
{
datapath0();
grafcet0();
printf("X0 = %d,X1 = %d,X2 = %d,X3 = %d,X4 = %d,X5 = %d,X6 = %d\n",X0 ,X1 ,X2 ,X3 ,X4 ,X5 ,X6 );
}
}
void grafcet0()
{

	if((X0 == 1) && (R0))
	{
		X0 = 0;
		X1 = 1;
		return;
	}

	if((X1 == 1) && (R1))
	{
		X1 = 0;
		X2 = 1;
		return;
	}

	if((X2 == 1) && (X24))
	{
		X2 = 0;
		X3 = 1;
		X4 = 1;
		return;
	}

	if((X3 == 1)&& (X5 == 1))
	{
		X3 = 0;
		X5 = 0;
		X6 = 1;
		return;
	}

	if((X4 == 1) && (R4))
	{
		X4 = 0;
		X5 = 1;
		return;
	}

	if((X6 == 1) && (R6))
	{
		X6 = 0;
		X0 = 1;
		return;
	}

}
void grafcet2()
{

	if((X20 == 1) && (X2))
	{
		X20 = 0;
		X21 = 1;
		return;
	}

	if((X21 == 1) && (R21))
	{
		X21 = 0;
		X22 = 1;
		return;
	}

	if((X22 == 1) && (X224))
	{
		X22 = 0;
		X23 = 1;
		return;
	}

	if((X23 == 1) && (X235))
	{
		X23 = 0;
		X24 = 1;
		return;
	}

	if((X24 == 1) && (1))
	{
		X24 = 0;
		X20 = 1;
		return;
	}

}
void grafcet22()
{

	if((X220 == 1) && (R220))
	{
		X220 = 0;
		X221 = 1;
		return;
	}

	if((X221 == 1) && (R221))
	{
		X221 = 0;
		X222 = 1;
		return;
	}

	if((X222 == 1) && (R222))
	{
		X222 = 0;
		X223 = 1;
		return;
	}

	if((X223 == 1) && (R223))
	{
		X223 = 0;
		X224 = 1;
		return;
	}

	if(X224 == 1)
	{
		if(x<0)
		{
			X224 = 0;
			X220 = 1;
		}
		else if(x>=0)
		{
			X224 = 0;
			X222 = 1;
		}
		return;
	}

}
void grafcet23()
{

	if((X230 == 1) && (X23))
	{
		X230 = 0;
		X231 = 1;
		return;
	}

	if(X231 == 1)
	{
		if(R231_1)
		{
			X231 = 0;
			X232 = 1;
		}
		else if(R231_2)
		{
			X231 = 0;
			X234 = 1;
		}
		return;
	}

	if((X232 == 1) && (R232))
	{
		X232 = 0;
		X233 = 1;
		return;
	}

	if(X233 == 1 && (R233))
	{
		X233 = 0;
		X235 = 1;
		return;
	}

	if(X234 == 1 && (R234))
	{
		X234 = 0;
		X235 = 1;
		return;
	}


	if((X235 == 1) && (1))
	{
		X235 = 0;
		X230 = 1;
		return;
	}

}
void datapath0()
{
if(X0 == 1)
action0();
if(X1 == 1)
action1();
if(X2 == 1)
action2();
if(X3 == 1)
action3();
if(X4 == 1)
action4();
if(X5 == 1)
action5();
if(X6 == 1)
action6();
}
void datapath2()
{
if(X20 == 1)
action20();
if(X21 == 1)
action21();
if(X22 == 1)
action22();
if(X23 == 1)
action23();
if(X24 == 1)
action24();
}
void datapath22()
{
if(X220 == 1)
action220();
if(X221 == 1)
action221();
if(X222 == 1)
action222();
if(X223 == 1)
action223();
if(X224 == 1)
action224();
}
void datapath23()
{
if(X230 == 1)
action230();
if(X231 == 1)
action231();
if(X232 == 1)
action232();
if(X233 == 1)
action233();
if(X234 == 1)
action234();
if(X235 == 1)
action235();
}
void action0()
{
printf("action0 activate !!\n");
}
void action1()
{
printf("action1 activate !!\n");
}
void action2()
{
printf("action2 activate !!\n");
datapath2();
grafcet2();
printf("X20 = %d,X21 = %d,X22 = %d,X23 = %d,X24 = %d\n",X20 ,X21 ,X22 ,X23 ,X24 );
}
void action3()
{
printf("action3 activate !!\n");
}
void action4()
{
printf("action4 activate !!\n");
}
void action5()
{
printf("action5 activate !!\n");
}
void action6()
{
printf("action6 activate !!\n");
}
void action20()
{
printf("action20 activate !!\n");
}
void action21()
{
printf("action21 activate !!\n");
}
void action22()
{
printf("action22 activate !!\n");
datapath22();
grafcet22();
printf("X220 = %d,X221 = %d,X222 = %d,X223 = %d,X224 = %d\n",X220 ,X221 ,X222 ,X223 ,X224 );
}
void action23()
{
printf("action23 activate !!\n");
datapath23();
grafcet23();
printf("X230 = %d,X231 = %d,X232 = %d,X233 = %d,X234 = %d,X235 = %d\n",X230 ,X231 ,X232 ,X233 ,X234 ,X235 );
}
void action24()
{
printf("action24 activate !!\n");
}
void action220()
{
printf("action220 activate !!\n");
}
void action221()
{
printf("action221 activate !!\n");
}
void action222()
{
printf("action222 activate !!\n");
}
void action223()
{
printf("action223 activate !!\n");
}
void action224()
{
printf("action224 activate !!\n");
}
void action230()
{
printf("action230 activate !!\n");
}
void action231()
{
printf("action231 activate !!\n");
}
void action232()
{
printf("action232 activate !!\n");
}
void action233()
{
printf("action233 activate !!\n");
}
void action234()
{
printf("action234 activate !!\n");
}
void action235()
{
printf("action235 activate !!\n");
}

File: Example/Grafcet.py
class Main():
	def __init__(self):
		self.X0 = self.X20 = self.X220 = self.X230 = 1
		self.X1 = self.X2 = self.X21 = self.X22 = self.X221 = self.X222 = self.X223 = self.X224 = self.X23 = self.X231 = self.X232 = self.X233 = self.X234 = self.X235 = self.X24 = self.X3 = self.X4 = self.X5 = self.X6 = 1
		print("X0 = %d,X1 = %d,X2 = %d,X3 = %d,X4 = %d,X5 = %d,X6 = %d"%(self.X0 ,self.X1 ,self.X2 ,self.X3 ,self.X4 ,self.X5 ,self.X6 ))
		while(True):
			self.datapath0()
			self.grafcet0()
			print("X0 = %d,X1 = %d,X2 = %d,X3 = %d,X4 = %d,X5 = %d,X6 = %d"%(self.X0 ,self.X1 ,self.X2 ,self.X3 ,self.X4 ,self.X5 ,self.X6 ))
	def grafcet0(self):

		if self.X0 == 1 and R0:
			self.X0 = 0
			self.X1 = 1
			return


		if self.X1 == 1 and R1:
			self.X1 = 0
			self.X2 = 1
			return


		if self.X2 == 1 and X24:
			self.X2 = 0
			self.X3 = 1
			self.X4 = 1
			return

		if self.X3 == 1 and self.X5 == 1 :
			self.X3 = 0
			self.X5 = 0
			self.X6 = 1
			return

		if self.X4 == 1 and R4:
			self.X4 = 0
			self.X5 = 1
			return


		if self.X6 == 1 and R6:
			self.X6 = 0
			self.X0 = 1
			return

	def grafcet2(self):

		if self.X20 == 1 and X2:
			self.X20 = 0
			self.X21 = 1
			return


		if self.X21 == 1 and R21:
			self.X21 = 0
			self.X22 = 1
			return


		if self.X22 == 1 and X224:
			self.X22 = 0
			self.X23 = 1
			return


		if self.X23 == 1 and X235:
			self.X23 = 0
			self.X24 = 1
			return


		if self.X24 == 1 and 1:
			self.X24 = 0
			self.X20 = 1
			return

	def grafcet22(self):

		if self.X220 == 1 and R220:
			self.X220 = 0
			self.X221 = 1
			return


		if self.X221 == 1 and R221:
			self.X221 = 0
			self.X222 = 1
			return


		if self.X222 == 1 and R222:
			self.X222 = 0
			self.X223 = 1
			return


		if self.X223 == 1 and R223:
			self.X223 = 0
			self.X224 = 1
			return


		if self.X224 == 1:
			if x<0:
				self.X224 = 0
				self.X220 = 1
			elif x>=0:
				self.X224 = 0
				self.X222 = 1
			return
	def grafcet23(self):

		if self.X230 == 1 and X23:
			self.X230 = 0
			self.X231 = 1
			return


		if self.X231 == 1:
			if R231_1:
				self.X231 = 0
				self.X232 = 1
			elif R231_2:
				self.X231 = 0
				self.X234 = 1
			return

		if self.X232 == 1 and R232:
			self.X232 = 0
			self.X233 = 1
			return


		if self.X233 == 1 and R233:
			self.X233 = 0
			self.X235 = 1
			return
		if self.X234 == 1 and R234:
			self.X234 = 0
			self.X235 = 1
			return

		if self.X235 == 1 and 1:
			self.X235 = 0
			self.X230 = 1
			return

	def datapath0(self):
		if self.X0 == 1:
			self.action0()
		if self.X1 == 1:
			self.action1()
		if self.X2 == 1:
			self.action2()
		if self.X3 == 1:
			self.action3()
		if self.X4 == 1:
			self.action4()
		if self.X5 == 1:
			self.action5()
		if self.X6 == 1:
			self.action6()
	def datapath2(self):
		if self.X20 == 1:
			self.action20()
		if self.X21 == 1:
			self.action21()
		if self.X22 == 1:
			self.action22()
		if self.X23 == 1:
			self.action23()
		if self.X24 == 1:
			self.action24()
	def datapath22(self):
		if self.X220 == 1:
			self.action220()
		if self.X221 == 1:
			self.action221()
		if self.X222 == 1:
			self.action222()
		if self.X223 == 1:
			self.action223()
		if self.X224 == 1:
			self.action224()
	def datapath23(self):
		if self.X230 == 1:
			self.action230()
		if self.X231 == 1:
			self.action231()
		if self.X232 == 1:
			self.action232()
		if self.X233 == 1:
			self.action233()
		if self.X234 == 1:
			self.action234()
		if self.X235 == 1:
			self.action235()
	def action0(self):
		print("action0 activate !!\n")
	def action1(self):
		print("action1 activate !!\n")
	def action2(self):
		print("action2 activate !!\n")
		self.datapath2()
		self.grafcet2()
		print("X20 = %d,X21 = %d,X22 = %d,X23 = %d,X24 = %d"%(self.X20,self.X21,self.X22,self.X23,self.X24))
	def action3(self):
		print("action3 activate !!\n")
	def action4(self):
		print("action4 activate !!\n")
	def action5(self):
		print("action5 activate !!\n")
	def action6(self):
		print("action6 activate !!\n")
	def action20(self):
		print("action20 activate !!\n")
	def action21(self):
		print("action21 activate !!\n")
	def action22(self):
		print("action22 activate !!\n")
		self.datapath22()
		self.grafcet22()
		print("X220 = %d,X221 = %d,X222 = %d,X223 = %d,X224 = %d"%(self.X220,self.X221,self.X222,self.X223,self.X224))
	def action23(self):
		print("action23 activate !!\n")
		self.datapath23()
		self.grafcet23()
		print("X230 = %d,X231 = %d,X232 = %d,X233 = %d,X234 = %d,X235 = %d"%(self.X230,self.X231,self.X232,self.X233,self.X234,self.X235))
	def action24(self):
		print("action24 activate !!\n")
	def action220(self):
		print("action220 activate !!\n")
	def action221(self):
		print("action221 activate !!\n")
	def action222(self):
		print("action222 activate !!\n")
	def action223(self):
		print("action223 activate !!\n")
	def action224(self):
		print("action224 activate !!\n")
	def action230(self):
		print("action230 activate !!\n")
	def action231(self):
		print("action231 activate !!\n")
	def action232(self):
		print("action232 activate !!\n")
	def action233(self):
		print("action233 activate !!\n")
	def action234(self):
		print("action234 activate !!\n")
	def action235(self):
		print("action235 activate !!\n")
if __name__ == '__main':
	Main()